game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/FiringGroupFilter.java

```diff
@@ -0,0 +1,8 @@
+package games.strategy.triplea.delegate.battle.steps.fire;
+
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.List;
+import java.util.function.Function;
+
+/** Converts the BattleState into a list of FiringGroups */
+public interface FiringGroupFilter extends Function<BattleState, List<FiringGroup>> {}
```

Comment:
Couple questions:
- Could you explain why an interface "alias" is helpful?
- Is the output list ordered?

A filter will typically exclude or include items. My presumption is that this is more of a sieve, that we are not losing any elements but instead are grouping them. Is that correct? If so, perhaps this would be better named as a `Splitter`, or as a `Sorter`.  OTOH - If we do not need an interface alias, then the naming problem would drop away.

Reply:
For consideration, [Effective Java 3](https://github.com/muhdkhokhar/test/blob/master/Joshua%20Bloch%20-%20Effective%20Java%20(3rd)%20-%202018.pdf), page 200:


![Screenshot from 2020-10-03 12-34-53](https://user-images.githubusercontent.com/12397753/95000358-f0d88900-0574-11eb-9695-664e1bd07bf2.png)

![Screenshot from 2020-10-03 12-36-30](https://user-images.githubusercontent.com/12397753/95000380-1bc2dd00-0575-11eb-80e6-26751a078c3f.png)

![Screenshot from 2020-10-03 12-34-35](https://user-images.githubusercontent.com/12397753/95000357-f03ff280-0574-11eb-8056-698c121fb20a.png)